buildbot/master/files/config/factories.py

```diff
@@ -103,6 +103,86 @@ def make_step(self, command):
         return step_class(**step_kwargs)
 
 
+class StepsYAMLParsingStep(buildstep.BuildStep):
+    """
+    Step which resolves the in-tree YAML and dynamically add test steps
+    """
+    yaml_path = ""
+
```

Comment:
We should set `haltOnFailure = True` and `flunkOnFailure = True` here, just as we do for `BadConfigurationStep`.